[feat] 대기 상세 뷰 구현

Doreminwoo opened the pull request:
## Work Description ✏️
- 대기 상세 화면 구현!

## Screenshot 📸
<img src="https://github.com/DOSOPT-CDS-TABLING/Tabling-Android/assets/86788873/2e8b642e-a929-4446-ab82-79cf3932b80a" width="360"/>

Doreminwoo committed:
[add] #6 Icon 파일 추가

Doreminwoo committed:
[add] #3 초기 파일 추가

Doreminwoo committed:
[mod] #6 themes Button Style 생성

Doreminwoo committed:
[mod] #6 string WaitingDetail에 사용하는 문자

Doreminwoo committed:
[add] #6 shape,bg 파일 생성

Doreminwoo committed:
[add] #6 Toolbar Menu Item

Doreminwoo committed:
[feat] #6 WaitingDetail UI 구현

Doreminwoo committed:
[add] #6 DummyData 추가

Doreminwoo committed:
[add] #6 Response 추가

Doreminwoo committed:
[feat] #6 DataBinding 연결

Doreminwoo committed:
[chore] #6 ktlint 적용

jihyunniiii commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,6 +15,9 @@
         android:theme="@style/Theme.TablingAndroid"
         android:usesCleartextTraffic="true"
         tools:targetApi="31">
+        <activity
+            android:name=".presentation.waiting.WaitingDetailActivity"
+            android:exported="false" />
```
```suggestion
        <activity
            android:name=".presentation.waiting.WaitingDetailActivity"
            android:exported="false"
            android:screenOrientation="portrait"
            tools:ignore="LockedOrientationActivity" />
```

추가해주셍요

Doreminwoo replied:
아 이거 모든 액티비티마다 적어야 되는건가요???

jihyunniiii replied:
네네 ㅠㅠ 다 적용해줘야함니다 ㅠ 제가 말을 안 해줬네요 제송티비

jihyunniiii commented on app/src/main/java/org/sopt/tabling/presentation/waiting/WaitingViewModel.kt:
```diff
@@ -0,0 +1,27 @@
+        MutableLiveData<ResponseWaitingDetailDto.WaitingDetailData>()
+    val waitingDetail: LiveData<ResponseWaitingDetailDto.WaitingDetailData> = _waitingDetail
+
+    init {
```
짱이다 !!!!!!!!!

Doreminwoo replied:
감삼둥!

jihyunniiii commented on app/src/main/res/drawable/ic_back_button_gray_600_45px.xml:
요거 뒤에 px 지워주고 그냥 숫자만 써주셍요 전체적으로 다 !

Doreminwoo replied:
넵

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:menu="@menu/item_waiting_detail_tb"
```
이거 그 새로고침 이미지 넣기 위해 한 건가요??? menu니까 네이밍은 menu_waiting_detail_tb로 하는 게 좋을 것 같구,,,, 이미지가 왤케 작아 보일까요? 기분탓인가,,, 한 번만 확인 부탁해염

Doreminwoo replied:
넵 이미지는 다시 새로 해봤는데도 크기가 똑같아요!

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+            app:titleTextAppearance="@style/TextAppearance.H2"
+            app:titleTextColor="@color/gray_800" />
+
+        <TextView
+            android:id="@+id/tv_waiting_detail_ing"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="17dp"
+            android:layout_marginTop="5dp"
+            android:text="@string/waiting_ing"
+            android:textColor="@color/gray_300"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/tb_waiting_detail" />
```
TextAppearance 지정해주세용

Doreminwoo replied:
넵!

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="9dp"
+            android:layout_marginEnd="6dp"
```
여기 마진 16인 듯요??? 확인 부탁해용

Doreminwoo replied:
헉 그러네요 !

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+
+        <Button
+            android:id="@+id/btn_waiting_detail_call"
+            style="@style/Style.Material3.Button.Outlined.Custom"
```
야무지다 야무져

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+                app:layout_constraintTop_toBottomOf="@id/tv_front_waiting" />
+        </androidx.constraintlayout.widget.ConstraintLayout>
+
+        <Button
+            android:id="@+id/btn_waiting_detail_call"
+            style="@style/Style.Material3.Button.Outlined.Custom"
+            android:layout_width="wrap_content"
+            android:layout_height="0dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="20dp"
+            android:text="@string/call"
+            app:icon="@drawable/ic_call_gray_500_24px"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/tv_waiting_detail_number" />
+
+        <Button
+            android:id="@+id/btn_waiting_detail_location"
+            style="@style/Style.Material3.Button.Outlined.Custom"
+            android:layout_width="wrap_content"
+            android:layout_height="0dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="20dp"
+            android:text="@string/location"
+            app:icon="@drawable/ic_location_gray_500_24px"
+            app:layout_constraintStart_toEndOf="@id/btn_waiting_detail_call"
+            app:layout_constraintTop_toBottomOf="@id/tv_waiting_detail_number" />
+
+        <Button
+            android:id="@+id/btn_waiting_detail_store"
+            style="@style/Style.Material3.Button.Outlined.Custom"
+            android:layout_width="wrap_content"
+            android:layout_height="0dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="20dp"
```
요 나란히 있는 버튼 3개를 구현할 때 저라면 기기 대응을 생각해서 체인을 걸어줄 것 같아요 !
기기에 맞춰 버튼 사이 공백 값을 동적으로 유지하기 위해서요 !
제 뷰에도 요런 버튼 3개 짜리가 있는데 제가 피알 올리면 어떤 식으로 구현했는지 한 번 참고하면 좋을 것 같네염 !!

Doreminwoo replied:
헉 그러네요 마진 패딩처리하느라 고생했는데 체인으로 묶어주니깐 균등하게 정렬되네요!
일단은 수정해뒀고 피알올리시면 참고하겠습니다!

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+            android:id="@+id/view_waiting_horizontal_3dp"
+            android:layout_width="0dp"
+            android:layout_height="3dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="22dp"
+            android:layout_marginEnd="16dp"
```
```suggestion
            android:layout_marginHorizontal="16dp"
            android:layout_marginTop="22dp"
```

Doreminwoo replied:
오홍 저 속성 하나로 코드 한줄이 줄어드는군요
동일한 경우에는 사용하면 좋네요!

jihyunniiii commented on app/src/main/res/layout/activity_waiting_detail.xml:
```diff
@@ -0,0 +1,350 @@
+            android:layout_width="0dp"
+            android:layout_height="1dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="22dp"
```
8dp 인 듯요???

Doreminwoo replied:
허걱 그러네요..
제가 조금 꼼꼼하지가 못하네요 ㅋㅎㅋㅎㅋㅎㅋㅋ

jihyunniiii commented on app/src/main/res/values/themes.xml:
```diff
@@ -35,6 +35,24 @@
         <item name="cornerRadius">6dp</item>
     </style>
 
+    <style name="Style.Material3.Button.Outlined" parent="Widget.Material3.Button.OutlinedButton">
+        <item name="cornerRadius">0dp</item>
+        <item name="android:padding">0dp</item>
+        <item name="android:drawablePadding">0dp</item>
+    </style>
+
+    <style name="Style.Material3.Button.Outlined.Custom" parent="Style.Material3.Button.Outlined">
+        <item name="cornerRadius">8dp</item>
+        <item name="iconTint">@null</item>
+        <item name="iconSize">@null</item>
+        <item name="backgroundColor">@color/white</item>
+        <item name="strokeColor">@color/gray_100</item>
+        <item name="android:paddingStart">16dp</item>
+        <item name="android:paddingEnd">29dp</item>
+        <item name="android:textColor">@color/gray_600</item>
+        <item name="android:textAppearance">@style/TextAppearance.Caption1.SemiBold</item>
+    </style>
+
```
아 야무지다 넘 잘한다 체고다

Doreminwoo replied:
감사합니다!!
다 덕분입니다!

jihyunniiii reviewed (commented):
수고했서용 !!!! 진짜 잘한다 !

Doreminwoo committed:
[chore] #6 코드리뷰 적용

HAJIEUN02 reviewed (approved):
왤케 잘하시는 거에욥ㅜ.ㅜ

Doreminwoo committed:
[chore] Develop Mager Confilcts 해결

Merge branch 'develop' into feat-waiting-detail-view

# Conflicts:
#	app/src/main/AndroidManifest.xml
#	app/src/main/res/values/strings.xml
